nsiqcppstyle_exe.py

```diff
@@ -217,6 +228,8 @@ def main(argv=None):
         nsiqcppstyle_reporter.PrepareReport(outputPath,
                                             _nsiqcppstyle_state.output_format)
         analyzedFiles = []
+        filterManager = None
+        filter = None
```

Comment:
Any particular reason for these 2 lines?

Reply:
Nothing gets past you, huh? ;-)

I added these because, without them, PyCharm's built-in CLang-Tidy "Problems" tab shows:

```
Local variable 'filter' might be referenced before assignment (line 315)
```

Here's the problematic line:

```
        nsiqcppstyle_reporter.ReportSummaryToScreen(analyzedFiles,
                                                    _nsiqcppstyle_state, filter)
```

Here's a highly redacted version of `main`:

```
def main(argv=None):
    ...
    try:
        ...
        targetPaths = GetRealTargetPaths(args)
        if len(targetPaths) == 0:
            ShowMessageAndExit("No target paths provided")
        ...
        #filter = None

        for targetPath in targetPaths:
            ...
            # Get Active Filter
            filterManager = FilterManager(filefilterPath, filterStringList, extLangMapCopy,
                                          varMap, filterScope)
            ...
            filter = filterManager.GetActiveFilter()
            ...
        nsiqcppstyle_reporter.ReportSummaryToScreen(analyzedFiles,
                                                    _nsiqcppstyle_state, filter)
        ...
```

Because we know that `targetPaths` is not empty (if it were, we'd exit with an error), we know we will enter the `for` loop and will assign a reference to `filter`.  Clang-Tidy, however, cannot assert that, and therefore warns us that `filter` in the `ReportSummaryToScreen` call might be referenced before being assigned to.  I added the `filter = None` line to remove the warning (there is no such need for the `filterManager = None` statement, I have removed that).